### PR TITLE
fixing IDENTITY-4531

### PR DESCRIPTION
**Pull request by mefarazath**
fix for https://wso2.org/jira/browse/IDENTITY-4531

* moving the SaaS enabled check to the authorizationGrantHandler validateGrant() method 
* removing the SaaS enabled check from PasswordGrantHandler validateGrant() method

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 1]
```diff
@@ -492,6 +503,36 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
...
+                    clientId, e);
+        }
+
+        String serviceProviderName = serviceProvider.getApplicationName();
+        // FIX for IDENTITY-4531
+        // if the service provider name is returned as "default" it means that a valid service provider for
+        // the given client ID was not found in the tenantDomain sent in the request
+        if (DEFAULT_SP_NAME.equals(serviceProviderName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Valid Service provider not found for client id "+clientId +
```
Comment (cdwijayarathna) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Leave spaces between Strings and +.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 2]
```diff
@@ -492,6 +503,36 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
...
+                        " and tenant domain "+tenantDomain);
+            }
+            return false;
+        }
+
+        if (!serviceProvider.isSaasApp() && !userTenantDomain.equals(tenantDomain)){
```
Comment (cdwijayarathna) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Fix the formatting

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 3]
```diff
@@ -492,6 +503,36 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx)
...
+            if(log.isDebugEnabled()) {
+                log.debug("Non-SaaS service provider tenant domain is not same as user tenant domain; " +
```
Comment (cdwijayarathna) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: What is the reason for having ; and != in error message?

Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: The debug log was already in PasswordGrantHandler, I will change the log to reflect the idea clearly

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java [hunk 4]
```diff
@@ -41,14 +38,21 @@
...
 import org.wso2.carbon.identity.oauth2.dto.OAuth2AccessTokenRespDTO;
+import org.wso2.carbon.identity.oauth2.internal.OAuth2ServiceComponentHolder;
 import org.wso2.carbon.identity.oauth2.model.AccessTokenDO;
-import org.wso2.carbon.identity.oauth2.token.OauthTokenIssuer;
 import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.token.OauthTokenIssuer;
 import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.sql.Timestamp;
```
Comment (cdwijayarathna) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java]: Why you have change the order of imports?

**Pull request closed without merging**